great_ape_safe/ape_api/ebtc.py

```diff
@@ -866,3 +872,211 @@ def authority_set_authority(self, new_authority):
                 EmptyBytes32,
                 delay + 1,
             )
+
+    #### ===== CDP OPS ===== ####
+
+    def _assert_collateral_balance(self, coll_amount):
+        total_coll_bal = self.collateral.balanceOf(self.safe.address)
+        assert total_coll_bal >= self.collateral.getSharesByPooledEth(coll_amount)
+
+    def _assert_debt_balance(self, debt_amount):
+        total_debt_bal = self.ebtc_token.balanceOf(self.safe.address)
+        assert total_debt_bal >= debt_amount
+        return total_debt_bal
+
+    def _assert_cdp_id_ownership(self, cdp_id):
+        cdp_safe_owned = self.sorted_cdps.getCdpsOf(self.safe.address)
+        assert cdp_id in cdp_safe_owned
+
+    def _hint_helper_values(self, coll_amount, debt_amount):
+        nicr = self.hint_helpers.computeNominalCR(coll_amount, debt_amount)
+
+        total_cdps = self.sorted_cdps.getSize()
+        hint, _, _ = self.hint_helpers.getApproxHint(
+            nicr, total_cdps, 42
+        )  # random seed 42
+
+        upper_hint, lower_hint = self.sorted_cdps.findInsertPosition(nicr, hint, hint)
+
+        return upper_hint, lower_hint
+
+    def open_cdp(self, coll_amount, target_cr):
+        """
+        @dev Opens a new cdp position.
+        @param coll_amount The total stETH collateral amount deposited for the specified Cdp.
+        @param target_cr The desired target collateral ratio in the cdp position.
+        """
+        # verify: is it coll balance available
+        self._assert_collateral_balance(coll_amount)
+
+        debt_balance_before = self.ebtc_token.balanceOf(self.safe.address)
+
+        # 1. collateral approval for BO
+        self.collateral.approve(self.borrower_operations.address, coll_amount)
+
+        # 2. decide borrow amount based on: collateral, feed price & CR
+        feed_price = self.price_feed.getPrice()
```

Comment:
After double checking, it seems that the `getPrice()` will no longer be supported: https://github.com/ebtc-protocol/ebtc/blob/be2e0ef382007b6c92db39afb44582dacb15a546/packages/contracts/contracts/PriceFeed.sol#L99

In addition, this function wouldn't have been accurate given that it used to return the last good price and not the current price, for that `fetchPrice` was and is the best approach. It is a call function that reads, aggregates and stores the current price as needed. I think for our purposes we can use a `.call()` to sim and obtain its current return value.

Reply:
good catch, def makes sense to use current and not last price cached

modified in 8c964637eb4c7c78a476cf14bd997c26abebe675